pkg/lease/lease_controller.go

```diff
@@ -52,14 +55,21 @@ func (r *leaseUpdater) Start(ctx context.Context) {
 	wait.JitterUntilWithContext(context.TODO(), r.reconcile, time.Duration(r.leaseDurationSeconds)*time.Second, leaseUpdateJitterFactor, true)
 }
 
-func (r *leaseUpdater) reconcile(ctx context.Context) {
-	for _, f := range r.healthCheckFuncs {
-		if !f() {
-			// IF a healthy check fails, do not update lease.
-			return
-		}
+func (r *leaseUpdater) WithHubKubeConfig(config *rest.Config) {
+	r.hubKubeconfig = config
+}
+
+func (r *leaseUpdater) updateHubLease(ctx context.Context) error {
+	hubClient, err := kubernetes.NewForConfig(r.hubKubeconfig)
+	if err != nil {
+		return err
 	}
-	lease, err := r.kubeClient.CoordinationV1().Leases(r.leaseNamespace).Get(context.TODO(), r.leaseName, metav1.GetOptions{})
+
+	return r.updateLease(ctx, hubClient)
+}
+
+func (r *leaseUpdater) updateLease(ctx context.Context, client kubernetes.Interface) error {
+	lease, err := r.kubeClient.CoordinationV1().Leases(r.leaseNamespace).Get(ctx, r.leaseName, metav1.GetOptions{})
```

Comment:
What's the value of `leaseNamespae` here? On hub it should be cluster namespace, while on managed cluster, it should be the installation namespace of addon.

Reply:
that is right.